draft-pauly-add-resolver-discovery.md

```diff
@@ -146,7 +150,38 @@ See {{local-deployment}} for local deployment considerations.
 
 # Discovery of DoH Capabilities for Direct Resolvers
 
-- Companion DoH server
+Direct Resolvers can advertise a Companion DoH server that offers equivalent services and is controlled 
+by the same entity. To do this, a DNS server MUST return an HTTPSSVC record for the "resolver.arpa"
+domain with "dohip" set to a valid IP address and the "dohuri" key set to a valid DoH URI 
+template as with the Designated DoH Server HTTPSSVC record. The TLS certificate used with the
+DoH URI MUST have the IP addresses for each of its DNS endpoints, classic or DoH, within the 
+SubjectAlternativeName field to allow the client to verify ownership.
+
+If the client is configured to query a Direct Resolver, it MUST query that resolver for an SVCB record for the 
```

Comment:
Again, I think we can just say "it queries that resolver..."

Reply:
Noted.

Reply:
Take two: I actually don't know which statement you're referring to here. All the ones pictured in this snippet I believe are required:
- The MUST for IP address inclusion is necessary for the client to be able to use the TLS cert to verify ownership of the referring IP address as well as the DoH IP address (whether they're the same or different)
- The MUST in line 160 for the client to query the SVCB record is necessary to ensure Companion DoH servers are discovered. If a client doesn't always query this record for DNS server IP addresses it uses, then it will miss DoH use where it is available.

Reply:
I was thinking of line 160, reading that as "if you want to query for a direct resolver, you have to send a query", which is more of a tautology than a normative statement. We're just describing how the desired action is taken. MUSTs should be reserved for security checks or interop problems. If your system can be configured to query, and your system just doesn't do it, that's a silly bug, because you have a configuration option that does nothing.

Reply:
Oh I see. What I was trying to communicate here is "if you've been configured to use foo, you need to ask foo for its Companion server before too long" but I guess the SHOULD statement already covers that. I'll remove the MUST and rephrase.

Reply:
Done.